### PR TITLE
Decaffeinate

### PR DESCRIPTION
**Pull request by deancouch**
Cleanup

- decaffeinate
- travis update
- updated dependencies

### CODE REVIEW

File: src/array-to-integer.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+const _ = require('lodash');
+
+const fromIntegerArray = function(integerArray) {
+  return getValue(_.uniq(integerArray), 'number', (function(_this) {
+    return function(bitmap, value, index) {
+      if (value > 0) {
+        return bitmap.value += Math.pow(2, (Math.floor(value)) - 1);
```
Comment (vickvu) [inline src/array-to-integer.js:7]: You can remove unnecessary return
```suggestion
        bitmap.value += Math.pow(2, (Math.floor(value)) - 1);
```

File: src/array-to-integer.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+const _ = require('lodash');
+
+const fromIntegerArray = function(integerArray) {
+  return getValue(_.uniq(integerArray), 'number', (function(_this) {
+    return function(bitmap, value, index) {
```
Comment (vickvu) [inline src/array-to-integer.js:5]: You can use `=>` here
```suggestion
  return getValue(_.uniq(integerArray), 'number', (bitmap, value, index) => {
```

File: src/array-to-integer.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      if (value > 0) {
+        return bitmap.value += Math.pow(2, (Math.floor(value)) - 1);
+      }
+    };
+  })(this));
```
Comment (vickvu) [inline src/array-to-integer.js:10]:
```suggestion
    });
```

File: src/array-to-integer.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+const fromBooleanArrayLE = function(booleanArray) {
+  return getValue(booleanArray, 'boolean', function(bitmap, value, index) {
```
Comment (vickvu) [inline src/array-to-integer.js:14]: and here
```suggestion
  return getValue(booleanArray, 'boolean', (bitmap, value, index) => {
```

File: src/array-to-integer.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    if (value) {
+      return bitmap.value += Math.pow(2, index);
```
Comment (vickvu) [inline src/array-to-integer.js:16]: here as well
```suggestion
      bitmap.value += Math.pow(2, index);
```

File: src/array-to-integer.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    }
+  });
+};
+
+const fromBooleanArrayBE = function(booleanArray) {
+  if (booleanArray instanceof Array) {
+    booleanArray.reverse();
+  }
+  return getValue(booleanArray, 'boolean', function(bitmap, value, index) {
```
Comment (vickvu) [inline src/array-to-integer.js:25]: same here
```suggestion
  return getValue(booleanArray, 'boolean', (bitmap, value, index) => {
```

File: src/array-to-integer.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    if (value) {
+      return bitmap.value += Math.pow(2, index);
```
Comment (vickvu) [inline src/array-to-integer.js:27]:
```suggestion
      bitmap.value += Math.pow(2, index);
```

File: src/array-to-integer.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+const validArray = function(array, valueType) {
+  var invalidValue;
+  if (!(array instanceof Array)) {
+    return false;
+  }
+  invalidValue = _(array).find(function(value) {
+    return typeof value !== valueType;
+  });
+  return invalidValue === void 0;
```
Comment (vickvu) [inline src/array-to-integer.js:40]: use findIndex and avoid lodash as array must be an instance of Array class, all in one line
```suggestion
  return array.findIndex((value) => typeof value !== valueType) < 0;
```

File: src/array-to-integer.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+const getValue = function(array, type, callback) {
+  var bitmap;
+  if (!validArray(array, type)) {
+    return 0;
+  }
+  bitmap = {
+    value: 0
+  };
+  _(array).forEach(function(value, index) {
```
Comment (vickvu) [inline src/array-to-integer.js:51]: No need to use lodash here (array must be an instance of Array class in `validArray` function)
```suggestion
  array.forEach((value, index) => {
```

File: src/array-to-integer.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+    return callback(bitmap, value, index);
```
Comment (vickvu) [inline src/array-to-integer.js:52]: Remove return
```suggestion
    callback(bitmap, value, index);
```

File: src/bitmap.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
+const _ = require('lodash');
```
Comment (vickvu) [inline src/bitmap.js:1]: no need to have lodash here
```suggestion
```

File: src/integer-to-array.js [hunk 12]
```diff
@@ -0,0 +1,43 @@
+const toIntegerArray = function(value) {
+  return getArray(value, function(result, character, index) {
+    if (character === '1') {
+      return result.push(index + 1);
```
Comment (vickvu) [inline src/integer-to-array.js:4]:
```suggestion
      result.push(index + 1);
```

File: src/integer-to-array.js [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+    }
+  });
+};
+
+const toBooleanArrayLE = function(value) {
+  return getArray(value, function(result, character, index) {
+    return result.push(character === '1' ? true : false);
```
Comment (vickvu) [inline src/integer-to-array.js:11]:
```suggestion
  return getArray(value, (result, character, index) => {
    result.push(character === '1' ? true : false);
```

File: src/integer-to-array.js [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+};
+
+const getArray = function(value, callback) {
+  var result;
+  if (!validNumber(value)) {
+    return [];
+  }
+  result = [];
+  Math.floor(value).toString(2).split('').reverse().forEach(function(character, index) {
+    return callback(result, character, index);
```
Comment (vickvu) [inline src/integer-to-array.js:36]:
```suggestion
  Math.floor(value).toString(2).split('').reverse().forEach((character, index) => {
    callback(result, character, index);
```

**Review by vickvu — CHANGES_REQUESTED** [review]
There are also a lot of `function(){` that can be changed to `()=>` and unnecessary `return` in the test files but they are not that important 😄

File: .travis.yml [hunk 15]
```diff
@@ -1,4 +1,6 @@
 language: node_js
 node_js:
-  - "0.10"
-script: "npm test"
\ No newline at end of file
+  - "12"
+  - "14"
+  - "16"
```
Comment (nizhu) [inline .travis.yml:5]: huuuge upgrade

**Review by nizhu — APPROVED** [review]
can replace vick with eslint

**Review by vickvu — APPROVED** [review]

**Pull request merged**